mlflow/store/_unity_catalog/registry/uc_oss_rest_store.py

```diff
@@ -0,0 +1,468 @@
+            full_name = get_full_name_from_sc(name, None)
+            [catalog_name, schema_name, model_name] = full_name.split(".")
+            req_body = message_to_json(
+                CreateModelVersion(
+                    model_name=model_name,
+                    catalog_name=catalog_name,
+                    schema_name=schema_name,
+                    source=source,
+                    run_id=run_id,
+                    comment=description,
+                )
+            )
+            model_version = self._call_endpoint(CreateModelVersion, req_body)
+            store = self._get_artifact_repo(model_version)
+            store.log_artifacts(local_dir=local_model_dir, artifact_path="")
+            endpoint, method = _METHOD_TO_INFO[FinalizeModelVersion]
+            final_endpoint = endpoint.replace("{full_name}", full_name).replace(
+                "{version}", str(model_version.version)
+            )
+            finalize_req_body = message_to_json(
+                FinalizeModelVersion(full_name=full_name, version=model_version.version)
+            )
+            registered_model_version = call_endpoint(
+                self.get_host_creds(),
+                endpoint=final_endpoint,
+                method=method,
+                json_body=finalize_req_body,
+                response_proto=self._get_response_from_method(FinalizeModelVersion),
+            )
+            return model_version_from_uc_oss_proto(registered_model_version)
+
+    def update_model_version(self, name, version, description):
+        full_name = get_full_name_from_sc(name, None)
+        [catalog_name, schema_name, model_name] = full_name.split(".")
+        req_body = message_to_json(
+            UpdateModelVersion(
+                full_name=full_name,
+                version=version,
+                comment=description,
+            )
+        )
+        endpoint, method = _METHOD_TO_INFO[UpdateModelVersion]
+        final_endpoint = endpoint.replace("{full_name}", full_name).replace(
+            "{version}", str(version)
+        )
+        registered_model_version = call_endpoint(
+            self.get_host_creds(),
+            endpoint=final_endpoint,
+            method=method,
+            json_body=req_body,
+            response_proto=self._get_response_from_method(UpdateModelVersion),
+        )
+        return model_version_from_uc_oss_proto(registered_model_version)
+
+    def transition_model_version_stage(self, name, version, stage, archive_existing_versions):
+        raise NotImplementedError("Method not implemented")
+
+    def delete_model_version(self, name, version):
+        full_name = get_full_name_from_sc(name, None)
+        req_body = message_to_json(DeleteModelVersion(full_name=full_name, version=version))
+        endpoint, method = _METHOD_TO_INFO[DeleteModelVersion]
+        final_endpoint = endpoint.replace("{full_name}", full_name).replace(
```

Comment:
seems like a function that could cover the version interpolation could have a default `None` for a `version` argument to handle replacing `version` optionally and `full_name` as a required replacement